horizons/gui/widgets/tooltip.py

```diff
@@ -97,103 +101,98 @@ def position_tooltip(self, event):
 		# and dropping these events seems to lead to the desired placements
-		def get_top(w):
-			return get_top(w.parent) if w.parent else w
-		top_pos = get_top(self).position
-		if top_pos == (0, 0):
-			return
+		# def get_top(w):
+		# 	return get_top(w.parent) if w.parent else w
+		# top_pos = get_top(self).position
+		# if top_pos == (0, 0):
+		# 	return
 
 		screen_width = horizons.globals.fife.engine_settings.getScreenWidth()
 		self.gui.y = widget_position[1] + y + 5
 		offset = x + 10
-		if (widget_position[0] + self.gui.size[0] + offset) > screen_width:
+
+		if (widget_position[0] + self.gui.size[0] + offset) > (screen_width / 2):
 			# right screen edge, position to the left of cursor instead
 			offset = x - self.gui.size[0] - 5
 		self.gui.x = widget_position[0] + offset
+
 		if not self.tooltip_shown:
 			self.show_tooltip()
-			#ExtScheduler().add_new_object(self.show_tooltip, self, run_in=0.3, loops=0)
-			self.tooltip_shown = True
 
 	def show_tooltip(self):
-		if not self.helptext:
-			return
+
 		if self.gui is None:
 			self.__init_gui()
 
-		#HACK: support icons in build menu
-		# Code below exists for the sole purpose of build menu tooltips showing
-		# resource icons. Even supporting that is a pain (as you will see),
-		# so if you think you need icons in other tooltips, maybe reconsider.
-		# [These unicode() calls brought to you by status icon tooltip code.]
+		self.gui.show()
+		self.tooltip_shown = True
+
+	def populate_tooltip(self):
+		"""This checks whether or not icons are to be shown and shows the if so.
+			This also sets the tooltips text.
+		"""
+		if not self.helptext:
+			return
+
+		# Specification looks like [[Buildmenu icon_index:amount (1:250 4:2 6:2)]]
 		buildmenu_icons = self.icon_regexp.findall(unicode(self.helptext))
 		# Remove the weird stuff before displaying text.
-		replaced = self.icon_regexp.sub('', unicode(self.helptext))
-		# Specification looks like [[Buildmenu 1:250 4:2 6:2]]
+		tooltip_text = self.icon_regexp.sub('', unicode(self.helptext))
+		self.label.max_width = self.MAX_LABEL_WIDTH
+
+
 		if buildmenu_icons:
 			hbox = HBox(position=(7, 5), padding=0)
 			for spec in buildmenu_icons[0].split():
 				(res_id, amount) = spec.split(':')
-				label = Label(text=amount+'  ')
+				resource_amount_label = Label(text=amount+'  ')
 				icon = Icon(image=get_res_icon_path(int(res_id)), size=(16, 16))
 				# For compatibility with FIFE 0.3.5 and older, also set min/max.
 				icon.max_size = icon.min_size = (16, 16)
-				hbox.addChildren(icon, label)
+				hbox.addChildren(icon, resource_amount_label)
 			hbox.adaptLayout()
-			# Now display the 16x16px "required resources" icons in the last line.
+			if hbox.width >= self.MAX_LABEL_WIDTH:
+				#self.MAX_LABEL_WIDTH = hbox.width
+				self.wrap_tt_text(buildmenu_icons, tooltip_text)
+				self.MAX_BG_WIDTH = hbox.width
+			else:
+				self.wrap_tt_text(buildmenu_icons, tooltip_text)
```

Comment:
Why is `self.MAX_BG_WIDTH` set here explicitly? Can you please explain this to me? It is also set in `self.wrap_tt_text()`. That's why I'm wondering. Maybe you could add a comment, why?